File: app/src/main/java/com/VegaSolutions/lpptransit/lppapi/responseobjects/TimetableWrapper.java
package com.VegaSolutions.lpptransit.lppapi.responseobjects;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class TimetableWrapper {

    private Station station;

    @SerializedName("route_groups")
    private List<RouteGroup> routeGroups;

    public Station getStation() {
        return station;
    }

    public List<RouteGroup> getRouteGroups() {
        return routeGroups;
    }

    public class Station {

        @SerializedName("ref_id")
        private String refId;

        private String name;

        public String getRefId() {
            return refId;
        }

        public String getName() {
            return name;
        }
    }

    public class RouteGroup {

        @SerializedName("route_group_number")
        private String routeGroupNumber;

        private List<Route> routes;

        public String getRouteGroupNumber() {
            return routeGroupNumber;
        }

        public List<Route> getRoutes() {
            return routes;
        }

        public class Route {

            private List<Timetable> timetable;
            private List<Station> stations;
            private String name;

            @SerializedName("parent_name")
            private String parentName;

            @SerializedName("group_name")
            private String groupName;

            @SerializedName("route_number_suffix")
            private String routeNumberSuffix;

            @SerializedName("route_number_prefix")
            private String routeNumberPrefix;

            public List<Timetable> getTimetable() {
                return timetable;
            }

            public List<Station> getStations() {
                return stations;
            }

            public String getName() {
                return name;
            }

            public String getParentName() {
                return parentName;
            }

            public String getGroupName() {
                return groupName;
            }

            public String getRouteNumberSuffix() {
                return routeNumberSuffix;
            }

            public String getRouteNumberPrefix() {
                return routeNumberPrefix;
            }

            public class Timetable {
                private int hour;
                private int[] minutes;

                @SerializedName("is_current")
                private boolean isCurrent;

                public int getHour() {
                    return hour;
                }

                public int[] getMinutes() {
                    return minutes;
                }

                public boolean isCurrent() {
                    return isCurrent;
                }
            }

            public class Station {

                @SerializedName("ref_id")
                private String refId;

                private String name;

                @SerializedName("order_no")
                private int orderNo;

                public String getRefId() {
                    return refId;
                }

                public String getName() {
                    return name;
                }

                public int getOrderNo() {
                    return orderNo;
                }
            }
        }

    }

}

File: app/src/main/java/com/VegaSolutions/lpptransit/data/lppapi/data/TimetableWrapper.kt
package com.VegaSolutions.lpptransit.data.lppapi.data

import com.google.gson.annotations.SerializedName

data class TimetableWrapper(
    @SerializedName("station")
    val station: Station,

    @SerializedName("route_groups")
    val routeGroups: List<RouteGroup>,
) {
    data class Station(
        @SerializedName("ref_id")
        val refId: String,

        @SerializedName("name")
        val name: String,
    )

    data class RouteGroup(
        @SerializedName("route_group_number")
        val routeGroupNumber: String,

        @SerializedName("routes")
        val routes: List<Route>,
    ) {
        data class Route(
            @SerializedName("timetable")
            val timetable: List<Timetable>,

            @SerializedName("stations")
            val stations: List<Station>,

            @SerializedName("name")
            val name: String,

            @SerializedName("parent_name")
            val parentName: String,

            @SerializedName("group_name")
            val groupName: String,

            @SerializedName("route_number_suffix")
            val routeNumberSuffix: String,

            @SerializedName("route_number_prefix")
            val routeNumberPrefix: String,

            @SerializedName("is_garage")
            val isGarage: Boolean,
        ) {
            data class Timetable(
                @SerializedName("hour")
                val hour: Int,

                @SerializedName("minutes")
                val minutes: List<Int>,

                @SerializedName("timestamp")
                val timestamp: String,

                @SerializedName("is_current")
                val isCurrent: Boolean,
            )

            data class Station(
                @SerializedName("ref_id")
                val refId: String,

                @SerializedName("name")
                val name: String,

                @SerializedName("order_no")
                val orderNo: Int,
            )
        }
    }
}
